client/sse.js

```diff
@@ -0,0 +1,74 @@
+import createDebug from 'debug';
+import { EventEmitter } from 'events';
+
+const debug = createDebug('sse-client');
+
+const Events = {
+  open: '#open'
+};
+
+const convertType = type => Events[type] || type;
+
+export default class SSEConnection extends EventEmitter {
+  constructor(url, options = {}) {
+    super();
+    this.options = options;
+    const { searchParams = {}, headers, timeout, ...config } = options;
+    if (debug.enabled) searchParams.debug = debug.namespace;
+    const streamUrl = createUrl(url, { searchParams });
+    this._connection = this.initialize(streamUrl, { headers, timeout, ...config });
+    this.once('open', ({ id }) => (this._id = id));
+    if (debug.enabled) this.on('message', e => debug('emitting event: %j', e));
```

Comment:
This kicks in once you execute the following snippet inside the browser console:
```js
localStorage.debug = 'sse-client';
```